From cd7a75caa320752369d7dd72cdf544fe2d0c26e5 Mon Sep 17 00:00:00 2001
From: Bin Yang <yangbin@rock-chips.com>
Date: Fri, 18 Oct 2019 11:14:00 +0800
Subject: [PATCH] usb: dwc2: Disable power down feature on Rockchip SoCs

Power down feature of DWC2 module integrated in Rockchip SoCs doesn't work
properly or needs some additional handling in PHY or SoC glue layer, so
disable it for now. Without disabling power down, DWC2 sporadically cannot
detect USB disconnect.

Change-Id: Ic46fdb7a000b9029727a6a46e4ca5399b98285e8
Signed-off-by: Bin Yang <yangbin@rock-chips.com>
---
 drivers/usb/dwc2/params.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/dwc2/params.c b/drivers/usb/dwc2/params.c
index 1cff88852798..26f0d46a5150 100644
--- a/drivers/usb/dwc2/params.c
+++ b/drivers/usb/dwc2/params.c
@@ -89,6 +89,7 @@ static void dwc2_set_rk_params(struct dwc2_hsotg *hsotg)
 	p->ahbcfg = GAHBCFG_HBSTLEN_INCR16 <<
 		GAHBCFG_HBSTLEN_SHIFT;
 	p->lpm = false;
+	p->power_down = 0;
 }
 
 static void dwc2_set_ltq_params(struct dwc2_hsotg *hsotg)
-- 
2.35.3

